### PR TITLE
Add ordered dictionary for quicker look up of Assessment packages in StatusHandler

### PR DESCRIPTION
**Pull request by feng-j678**
What is in this PR:
[X] Modify the nested for loop in set_package_assessment_status to use order dictionary for quicker look up
[X] unit test assessment order dictionary

### CODE REVIEW

**Review by kjohn-msft — DISMISSED** [review]

File: src/core/tests/Test_StatusHandler.py [hunk 1]
```diff
@@ -396,6 +396,32 @@ def test_sort_packages_by_classification_and_state(self):
...
+        patch_id = 'python-samba0_2:4.4.5+dfsg-2ubuntu5.4_Ubuntu_16.04'
+        expected_value = {'version': '2:4.4.5+dfsg-2ubuntu5.4', 'classifications': ['Critical'], 'name': 'python-samba0', 'patchId': 'python-samba0_2:4.4.5+dfsg-2ubuntu5.4_Ubuntu_16.04'}
+
+        status_handler = StatusHandler(self.runtime.env_layer, self.runtime.execution_config, self.runtime.composite_logger, self.runtime.telemetry_writer, self.runtime.vm_cloud_type)
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count_for_test)
+        status_handler.set_package_assessment_status(test_packages, test_package_versions, 'Critical')
+        self.assertIsNotNone(status_handler._StatusHandler__assessment_packages_map)
```
Comment (rane-rajasi) [inline src/core/tests/Test_StatusHandler.py:410]: I don't agree with checking a functions internal variables in asserts this way. My preference would be to check the output of the function that's being tested. Which in this instance would be the output of set_package_assessment_status() 

Adding @kjohn-msft to bring this to attention

**Review by rane-rajasi — CHANGES_REQUESTED** [review]
Comment inline

File: src/core/tests/Test_StatusHandler.py (hunk 1 above)
Comment (feng-j678) [reply src/core/tests/Test_StatusHandler.py:418]: add more assertions for checking if status_file has the new package chagnes

Comment (rane-rajasi) [reply src/core/tests/Test_StatusHandler.py:418]: We don't need to check StatusHandler's internal variables to verify if packages exist. Line 412 is reading contents from the status file, you can directly check if the packages exist within that file.
Please remove lines 418 to 423 and check for packages within status file using the status file read in line 412.

Please see lines 415, 416, 417 for reference.

**Review by rane-rajasi — CHANGES_REQUESTED** [review]
Comment inline

File: src/core/tests/Test_StatusHandler.py (hunk 1 above)
Comment (feng-j678) [reply src/core/tests/Test_StatusHandler.py]: done, the reference to internal variable is removed

**Review by kjohn-msft — APPROVED** [review]

**Review by rane-rajasi — APPROVED** [review]
Now that the py27 CI issue is fixed, let that PR go in first, so you can fetch latest from master into this PR, run py27 on this PR and then merge.

**Pull request merged**